### PR TITLE
ensure that sparse matrix shapes are always a tuple of int

### PR DESCRIPTION
**Pull request by grlee77**
It is currently possible to pass non-integers as the elements of the `shape` input to the sparse matrix creation routines.  When this is done, the `__init__` method should make sure these get converted to proper Python `int`.  This conversion to `int` is already done if the first argument is a shape, but was missing when `shape` was instead supplied via the keyword argument.

A remaining difference is that the existing `util.isshape` function still allows a shape such as `(3.0, 4.0)` to be converted to `(3, 4)` while SciPy would raise the following error:
`TypeError: 'float' object cannot be interpreted as an integer`

### CODE REVIEW

File: cupyx/scipy/sparse/compressed.py [hunk 1]
```diff
@@ -130,7 +130,9 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                              % (len(indptr), major + 1))
 
         self._descr = cusparse.MatDescriptor.create()
-        self._shape = shape
+        if not util.isshape(shape):
+            raise ValueError('invalid shape (must be a 2-tuple of int)')
+        self._shape = int(shape[0]), int(shape[1])
```
Comment (hvy) [inline cupyx/scipy/sparse/compressed.py]: If it is the case that this is only relevant if `shape` is supplied as an argument (that's not `None`), how about doing this processing at the top?

**Comment by hvy** [comment]
Thanks for the PR! I left a comment but it LGTM otherwise. Could you take a look?

**Comment by hvy** [comment]
jenkins, test this please.

File: cupyx/scipy/sparse/compressed.py (hunk 1 above)
Comment (grlee77) [reply cupyx/scipy/sparse/compressed.py]: I have moved it up to the top as suggested. Thanks!

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**